src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -56,7 +55,57 @@ public class DefaultHtmlGenerator : IHtmlGenerator
             IModelMetadataProvider metadataProvider,
             IUrlHelperFactory urlHelperFactory,
             HtmlEncoder htmlEncoder,
-            ClientValidatorCache clientValidatorCache)
+            ClientValidatorCache clientValidatorCache) : this(
+                antiforgery,
+                optionsAccessor,
+                metadataProvider,
+                urlHelperFactory,
+                htmlEncoder,
+                clientValidatorCache,
+                () => new ValidationAttributeProvider(optionsAccessor, metadataProvider, clientValidatorCache))
```

Comment:
Because `ValidationAttributeProvider` will `throw` if any of the arguments passed to the constructor are `null`. That leads to different failures than before e.g. would complain about `antiforgery` today and `metadataProvider` tomorrow if both were `null`.

But I'm probably overthinking back-compat in these error cases. Will do the simpler thingy.